picasso/src/main/java/com/squareup/picasso/MarkableInputStream.java

```diff
@@ -119,6 +129,9 @@ private void skip(long current, long target) throws IOException {
   }
 
   @Override public int read() throws IOException {
+    if (!allowExpire && (offset + 1 > limit)) {
+      setLimit(limit + limitIncrement);
```

Comment:
Ideally this limit doubles every time its hit. Otherwise you needlessly churn through memory and waste copies in the pathological degradation case that this is meant to fix.